lib/cors-anywhere.js

```diff
@@ -370,6 +371,12 @@ function getHandler(options, proxy) {
       res.end('The origin "' + origin + '" was not whitelisted by the operator of this proxy.');
       return;
     }
+    
+    if (corsAnywhere.targetWhitelist.length && corsAnywhere.targetWhitelist.indexOf(location.href) === -1) {
```

Comment:
What's the use case for this? This exact match is quite inflexible. If you use cors-anywhere as a library, then you can already enforce this kind of target restriction by comparing `req.url.slice(1)` with the target whitelist.

To learn more about potential implementation details to improve the design, see #78. See also the review feedback at #111.